File: stegochat-backend/src/main/java/stegochat/stegochat/service/UserService.java
package stegochat.stegochat.service;

import stegochat.stegochat.dto.UserDTO;
import stegochat.stegochat.entity.enums.OtpType;

import java.util.Optional;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

public interface UserService {

    void initiateRegistration(UserDTO userDTO);

    UserDTO completeRegistration(String email, String otp);

    UserDTO loginUser(String username, String password, HttpServletRequest request, HttpServletResponse response);

    UserDTO getUserProfile(HttpServletRequest request);

    Optional<String> getUserByUsername(String username);

    void changePassword(HttpServletRequest request, String currentPassword, String newPassword);

    void updatePhoneNumber(HttpServletRequest request, String newPhoneNumber);

    void initiateEmailUpdate(HttpServletRequest request, String newEmail);

    void verifyEmailUpdate(HttpServletRequest request, String otp); 
    
    void resendOtp(String email, OtpType otpType);

    void logout(HttpServletRequest request, HttpServletResponse response);
    
}

File: stegochat-frontend/src/services/userService.js
import api from "./api";

const UserService = {
    updateEmail: async (newEmail) => {
        const response = await api.put("/users/update-email", { newEmail });
        return response.data;
    },

    updatePhone: async (newPhoneNumber) => {
        const response = await api.put("/users/update-phone", { newPhoneNumber });
        return response.data;
    },

    changePassword: async (currentPassword, newPassword) => {
        const response = await api.put("/users/change-password", { currentPassword, newPassword });
        return response.data;
    },

    searchUsers: async (query) => {
        const response = await api.get(`/users/search?query=${query}`);
        return response.data;
    },

};

export default UserService;
